### DISCUSSION TITLE
Can Kornia work using latest Pytorch/Torchvision version (1.10.1) ?

### DISCUSSION THREAD

**Discussion by ParadoxRobotics** [kornia/kornia#1517] [Potential Issue] [answered]
Hi, 

I would like to use the library for pre-processing in my project. I'm currently using Pytorch 1.10.1 with CUDA 11.3  in my project and I don't want to change version. When installing Kornia I noticed that it try to download Pytorch 1.8.1 (which I don't want).
Is it possible to use Kornia with my current version of Pytorch / Torchvision ?

**Comment by edgarriba** [comment, marked as answer]
Does it happen the same if you already have pytorch 1.10.1 ? In principle kornia only download if there's no cached torch version.

**Reply by ParadoxRobotics** [reply]
Yes; I already have pytorch 1.10.1 on my computer (since the release).
Kornia still try to install 1.8.1 even with my current version.
I try the pip install and install from setup.py but the same happen (I abort the download and install in both case).
Is it a setup.py error ? or is it an Environment variable problem from my pytorch install ? 

(I install pytorch/torchvision (1.10.1 with cuda 11.3) with : 
pip3 install torch==1.10.1+cu113 torchvision==0.11.2+cu113 torchaudio==0.10.1+cu113 -f https://download.pytorch.org/whl/cu113/torch_stable.html)

**Reply by ParadoxRobotics** [reply]
I manage to install it, by forcing the version in the setup.py.
It work without problem with the "hello world" example.

**Reply by edgarriba** [reply]
weird -- is it then an issue from our side. In any case, I'll try to send a patch to properly update/test on 1.10.1

**Reply by edgarriba** [reply]
In principle our minimum torch version is 1.8.1
https://github.com/kornia/kornia/blob/master/setup.py#L29

I remember sometimes there are issues around the postfix `cu*` -- /cc @pmeier dealt with that at some point. Any ideas ?

**Reply by pmeier** [reply]
@ParadoxRobotics I can't reproduce. What Python version are you on? Are these your install steps?

```shell
$ pip install torch==1.10.1+cu113 torchvision==0.11.2+cu113 \
    -f https://download.pytorch.org/whl/cu113/torch_stable.html
$ pip install kornia
```

**Reply by ParadoxRobotics** [reply]
I'm on python 3.8, 
my install procedure is : 
pip3 install torch==1.10.1+cu113 torchvision==0.11.2+cu113 torchaudio==0.10.1+cu113 -f https://download.pytorch.org/whl/cu113/torch_stable.html
pip3 install git+https://github.com/kornia/kornia

**Reply by pmeier** [reply]
Still, can't reproduce in a clean environment. Could you paste your output?

**Reply by edgarriba** [reply]
a shoot in air, but the only difference I see is `pip3` ?

**Reply by ParadoxRobotics** [reply]
I try to reproduce the mishaps but without success on a fresh installation.
I don't have the original output anymore.
I suppose it was a problem coming from the pkg profile on my Ubuntu. 

Sorry for the inconvenience.

**Reply by edgarriba** [reply]
no prob :)
